matching.py

```diff
@@ -224,12 +224,12 @@ def maxquant_file_reader(filepath):
 
     return focused_maxquant_df
 
-def main(ftrs_filepath:str, csv_filepath:str):
+def main(ftrs_filepath:str, csv_filepath:str, RT_window:float, Sodium:bool, Potassium:bool, Anhydro:bool, DeAc:bool, Deacetyl_Anh:bool, Nude:bool, Decay:bool, Amidation:bool, Amidase:bool, Double_Anh:bool):
```

Comment:
Couple of things here:

* Please can `main` take data frames as arguments, rather than paths? This means if it's being called by a function that already has the data in memory, there's no need to reload.
* Can `main` be called `match` (or something more descriptive than `main`)?
* There's too many arguments to easily read - maybe a list of modifications would work better? e.g.

```
mod_list = ["Sodium", "Potassium", "Ketchup"]

if "Sodium" in mod_list:
      squiddly diddly diddly do stuff
```

I think there may be a better data framey way of handling this longer term i.e. a function that takes some masses (df) and mods (list/df) and returns the modified list.

Reply:
Yeah, I'll make those changes for tomorrow. 
I'll look into more elegant solution to having so many optional conditions

Reply:
Changes made and pushed to latest commit in this branch

I need to find away to make the ppm_tol variable in the calc_ppm_tol func setable by the user. I tried to assign it to a global variable and made a method to set the global variable but it wasn't working as intended. Will keep trying.

Reply:
Looking really nice! You'll need to pass `ppm_tol` via `data_analysis`, I think.

Also, I'd suggest more frequent commits with more specific messages e.g. "go from several Booleans to list".

Reply:
Generally avoid global variables. They're hard to keep track of.